packages/assets-controllers/src/token-prices-service/codefi-v2.ts

```diff
@@ -349,19 +364,23 @@ export class CodefiTokenPricesServiceV2
     currency: SupportedCurrency;
   }): Promise<Partial<TokenPricesByTokenAddress<Hex, SupportedCurrency>>> {
     const chainIdAsNumber = hexToNumber(chainId);
+    const ZERO_ADDRESS: `0x${string}` =
+      '0x0000000000000000000000000000000000000000';
 
     const url = new URL(`${BASE_URL}/chains/${chainIdAsNumber}/spot-prices`);
-    url.searchParams.append('tokenAddresses', tokenAddresses.join(','));
+    url.searchParams.append(
+      'tokenAddresses',
+      [ZERO_ADDRESS, ...tokenAddresses].join(','),
+    );
     url.searchParams.append('vsCurrency', currency);
+    url.searchParams.append('includeMarketData', 'true');
 
-    const pricesByCurrencyByTokenAddress: SpotPricesEndpointData<
-      Lowercase<Hex>,
-      Lowercase<SupportedCurrency>
-    > = await this.#tokenPricePolicy.execute(() =>
-      handleFetch(url, { headers: { 'Cache-Control': 'no-cache' } }),
-    );
+    const pricesByCurrencyByTokenAddress: AddressCryptoDataMap =
+      await this.#tokenPricePolicy.execute(() =>
+        handleFetch(url, { headers: { 'Cache-Control': 'no-cache' } }),
+      );
 
-    return tokenAddresses.reduce(
+    return [ZERO_ADDRESS, ...tokenAddresses].reduce(
```

Comment:
Why are we passing this now?

Reply:
same explanation , when a user has imported ERC20 tokens, the API call includes the zero address along with the addresses of all those tokens to get the market data of the native token